src/applications/disability-benefits/all-claims/submit-transformer.js

```diff
@@ -272,33 +270,31 @@ export function transform(formConfig, form) {
       return formData;
     }
     const clonedData = _.cloneDeep(formData);
-    const newVAFacilities = clonedData.vaTreatmentFacilities.map(facility => {
+    const newVAFacilities = clonedData.vaTreatmentFacilities.map(facility =>
       // Transform the related disabilities lists into an array of strings
-      const newFacility = _.set(
+      _.set(
         'treatedDisabilityNames',
         transformRelatedDisabilities(
           facility.treatedDisabilityNames,
           getClaimedConditionNames(formData),
         ),
         facility,
-      );
-      // If the day is missing, set it to the last day of the month because EVSS requires a day
-      const [year, month, day] = _.get(
-        'treatmentDateRange.to',
-        newFacility,
-        '',
-      ).split('-');
-      if (day && day === 'XX') {
-        newFacility.treatmentDateRange.to = moment(`${year}-${month}`)
-          .endOf('month')
-          .format('YYYY-MM-DD');
-      }
-      return newFacility;
-    });
+      ),
+    );
     clonedData.vaTreatmentFacilities = newVAFacilities;
     return clonedData;
   };
 
+  const transformSeparationPayDate = formData => {
+    if (!formData.separationPayDate) {
+      return formData;
+    }
+    const clonedData = _.cloneDeep(formData);
```

Comment:
Could we use a `_.set()` here instead? Not that it matters a _whole_ lot, but it's a bit more direct.